api/mapping/test_views.py

```diff
@@ -0,0 +1,161 @@
+                    self.restricted_dataset.id,
+                ],
+            )
+
+        # Assert user1 can't see public_dataset3
+        for obj in response_data:
+            self.assertNotEqual(obj.get("id"), self.public_dataset3.id)
+
+        # Add user2 to the request; this is not automatic
+        request.user = self.user2
+        # Authenticate the user2
+        force_authenticate(
+            request,
+            user=self.user2,
+            token=self.user2.auth_token,
+        )
+        # Get the response
+        response_data = self.view(request).data
+
+        # Assert user2 can only public_dataset1 and public_dataset2
+        for obj in response_data:
+            self.assertIn(
+                obj.get("id"),
+                [self.public_dataset1.id, self.public_dataset2.id],
+            )
+
+        # Assert user2 can't see public_dataset3
+        for obj in response_data:
+            self.assertNotEqual(obj.get("id"), self.public_dataset3.id)
+
+    def test_dataset_filtering(self):
+        # Make the request for the public_dataset1
+        request = self.factory.get(
+            f"api/datasets/", {"id__in": self.public_dataset1.id}
+        )
+        # Add user1 to the request; this is not automatic
+        request.user = self.user1
+        # Authenticate user1
+        force_authenticate(
+            request,
+            user=self.user1,
+            token=self.user1.auth_token,
+        )
+        # Get the response
+        response_data = self.view(request).data
+
+        # Assert only got public_dataset1
+        self.assertEqual(len(response_data), 1)
+        self.assertEqual(response_data[0].get("id"), self.public_dataset1.id)
+
+        # Make the request for the public_dataset3
+        request = self.factory.get(
+            f"api/datasets/", {"id__in": self.public_dataset3.id}
+        )
+        # Add user1 to the request; this is not automatic
+        request.user = self.user1
+        # Authenticate user1
+        force_authenticate(
+            request,
+            user=self.user1,
+            token=self.user1.auth_token,
+        )
+        # Get the response
+        response_data = self.view(request).data
+
+        # Assert response is empty
+        self.assertEqual(response_data, [])
+
+    def test_az_function_user_perm(self):
+        User = get_user_model()
+        az_user = User.objects.get(username=os.getenv("AZ_FUNCTION_USER"))
+        # Make the request for the Dataset
+        request = self.factory.get(f"api/datasets/")
+        # Add the user to the request; this is not automatic
+        request.user = az_user
+        # Authenticate az_user
+        force_authenticate(
+            request,
+            user=az_user,
+            token=az_user.auth_token,
+        )
+        # Get the response
+        response_data = self.view(request).data
+        # Assert az_user can see all datasets
+        for obj in response_data:
```

Comment:
Does this positively show `response_data` contains all the datasets, or simply that `response_data` is a subset of all datasets?